pkg/utils/filters/rewrite.go

```diff
@@ -0,0 +1,40 @@
+package filters
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+const OriginPathHeaderKey = "X-Rewrite-Original-Path"
+const OldResourceApiServerPrefixWithoutSlash = "/apis/clusterpedia.io/v1beta1/resources"
+const OldResourceApiServerPrefix = OldResourceApiServerPrefixWithoutSlash + "/"
+
+func WithRewriteFilter(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		oldPath := req.URL.EscapedPath()
+		if rewritePath, ok := urlPrefixRewrite(oldPath); ok {
+			req.URL.Path = rewritePath
+			req.Header.Set(OriginPathHeaderKey, oldPath)
+			klog.V(5).InfoS("request need rewrite", "oldPath", oldPath, "newPath", req.URL.EscapedPath())
+		} else {
+			klog.V(5).InfoS("request not need rewrite", "path", oldPath)
+		}
+
+		handler.ServeHTTP(w, req)
+	})
+}
+
+func urlPrefixRewrite(oldPath string) (string, bool) {
+	if strings.HasPrefix(oldPath, OldResourceApiServerPrefix) || oldPath == OldResourceApiServerPrefixWithoutSlash {
+		return "", false
+	}
+
+	rewritePath, err := url.JoinPath(OldResourceApiServerPrefix, oldPath)
```

Comment:
For path like /healthz, we probably don't want to be prefixed.

And since the only prefixes for kube resources are api and apis, we can just rewrite the prefixes for those two.

Reply:
@Iceber In origin implement the /healthz implement by `https://xxx/apis/clusterpedia.io/v1beta1/resources` , so we rewrite it too.

Reply:
We will provide paths for `/livez`,`/readyz` and other *clusterpedia apiserver application levels* in the future, their paths should not be under resources but under the root path, i don't think it should be required to have only apis/clusterpedia.io/* paths under the root path